browse/util/id_patterns.py

```diff
@@ -0,0 +1,323 @@
+r"""Patterns and functions to detect arXiv ids and Urls in text.
```

Comment:
random r at start of this line? or is it supposed to be there?

Reply:
It is the flag to python that the string should be escaped for regexes.  It is there because the comment has slashes and one of the linters was complaining that it might be a regex.

Reply:
My bad, the linter did complain :-) . I've reverted my previous change.